compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -10609,4 +10601,130 @@ protected void addTransactionInternalModuleImport() {
             env.enclPkg.symbol.imports.add(importDcl.symbol);
         }
     }
+
+    private void createMapFunctionForStepExpr(BLangXMLNavigationAccess xmlNavigation,
+                                              List<BLangXMLStepExtend> extensions) {
+        Location pos = xmlNavigation.pos;
+        ArrayList<BLangExpression> args = new ArrayList<>();
+        ArrayList<BLangExpression> filters = expandFilters(xmlNavigation.filters);
+
+        // xml/**/<elemName>
+        if (xmlNavigation.navAccessType == XMLNavigationAccess.NavAccessType.DESCENDANTS) {
+            args.add(createArrowFunctionForNavigation(pos, extensions, XML_GET_DESCENDANTS, filters));
+        } else {
+            // xml/*
+            args.add(createArrowFunctionForNavigation(pos, extensions, XML_INTERNAL_CHILDREN, filters));
+        }
+
+        BLangInvocation elements =
+                createLanglibXMLInvocation(pos, XML_ELEMENTS, xmlNavigation.expr, new ArrayList<>(),
+                        new ArrayList<>());
+        BLangInvocation invocationNode = createLanglibXMLInvocation(pos, XML_MAP, elements, args, new ArrayList<>());
+        result = rewriteExpr(invocationNode);
+    }
+
+    public BLangLambdaFunction createArrowFunctionForNavigation(Location pos, List<BLangXMLStepExtend> extensions,
+                                                                String func, ArrayList<BLangExpression> filters) {
```

Comment:
Is we can apply above suggestion let's relax this parameter to be List as well
```suggestion
                                                                String func, List<BLangExpression> filters) {
```